### PR TITLE
Sync polling

### PR DESCRIPTION
**Pull request by almogdepaz**
## Motivation
sync waits for layer tick to start syncing, this pr turns sync into using an independent timer
to allow better useabilty

them when this it's merged -->

## Changes
syncer will check sync status every "SyncInterval"

## Test Plan
same as before

### CODE REVIEW

**Comment by almogdepaz** [comment]
bors try

File: cmd/node/node.go [hunk 1]
```diff
@@ -510,7 +510,14 @@ func (app *SpacemeshApp) initServices(nodeID types.NodeId,
 		app.setupGenesis(processor, msh)
 	}
 
-	conf := sync.Configuration{Concurrency: 4, LayerSize: int(layerSize), LayersPerEpoch: layersPerEpoch, RequestTimeout: time.Duration(app.Config.SyncRequestTimeout) * time.Millisecond, Hdist: app.Config.Hdist, AtxsLimit: app.Config.AtxsPerBlock}
+	syncConf := sync.Configuration{Concurrency: 4,
+		LayerSize:      int(layerSize),
+		LayersPerEpoch: layersPerEpoch,
+		RequestTimeout: time.Duration(app.Config.SyncRequestTimeout) * time.Millisecond,
+		SyncInterval:   time.Duration(app.Config.SyncInterval) * 10,
```
Comment (gavraz) [inline cmd/node/node.go]: what units are you using? use time.Second or time.Millisecond...

File: config/config.go [hunk 2]
```diff
@@ -90,6 +90,8 @@ type BaseConfig struct {
 
 	SyncRequestTimeout int `mapstructure:"sync-request-timeout"` // the timeout for direct request in the sync
 
+	SyncInterval int `mapstructure:"sync-interval"`
```
Comment (gavraz) [inline config/config.go]: Specify the units of this interval (seconds probably).

File: sync/syncer.go [hunk 3]
```diff
@@ -104,14 +105,12 @@ type Syncer struct {
...
 	lValidator           LayerValidator
-	currentLayer         types.LayerID
-	currentLayerMutex    sync.RWMutex
 	validatingLayer      types.LayerID
 	validatingLayerMutex sync.Mutex
 	SyncLock             uint32
 	startLock            uint32
 	forceSync            chan bool
-	LayerCh              timesync.LayerTimer
+	syncTimer            *time.Ticker
```
Comment (gavraz) [inline sync/syncer.go:113]: Make sure the disadvantages of time.Ticker (aggregate delay, resources release) are no bother for your use case/you handle it.
In addition, you can use an interface here and then you will be able to test it better with a "ticker" mock.

File: sync/syncer.go [hunk 4]
```diff
@@ -224,17 +223,13 @@ func (s *Syncer) run() {
 			return
 		case <-s.forceSync:
 			go syncRoutine()
-		case layer := <-s.LayerCh:
-			s.currentLayerMutex.Lock()
-			s.currentLayer = layer
-			s.currentLayerMutex.Unlock()
-			s.Debug("sync got tick for layer %v", layer)
+		case <-s.syncTimer.C:
```
Comment (gavraz) [inline sync/syncer.go:226]: There couldn't be more than one syncRoutine right?

**Review by gavraz — CHANGES_REQUESTED** [review]

File: sync/syncer.go (hunk 3 above)
Comment (almogdepaz) [reply sync/syncer.go:113]: cool idea, maybe in a later revision when the need arrives, for now this supports all our current test cases

File: sync/syncer.go (hunk 4 above)
Comment (almogdepaz) [reply sync/syncer.go:226]: nope

File: config/config.go (hunk 2 above)
Comment (almogdepaz) [reply config/config.go]: cool ill add some documentation

**Comment by almogdepaz** [comment]
bors try

**Review by gavraz — APPROVED** [review]

**Comment by almogdepaz** [comment]
bors merge

**Pull request merged**